fs/bcachefs/fs-io.c

```diff
@@ -2306,12 +2306,6 @@ int bch2_truncate(struct user_namespace *mnt_userns,
 	s64 i_sectors_delta = 0;
 	int ret = 0;
 
-	/*
-	 * Don't update timestamps if we're not doing anything:
-	 */
-	if (iattr->ia_size == inode->v.i_size)
```

Comment:
This causes `generic/092` to fail as the truncate should be run as we've allocated some sectors beyond the `i_size`. I don't think we could use something like `iattr->ia_size == inode->v.i_blocks << 9` either, because the file may contain a hole. Some filesystems seem to have optimizations like this for zero length files, so I could change this check to something for zero length files if we want it.